// report-a-problem/js/report-a-problem.js
/** Start Report a Problem **/

/** Report a Broken Link **/
app.component('prmAlmaViewitAfter', {
	bindings: { parentCtrl: '<' },
    controller: 'prmAlmaViewitAfterController',
    templateUrl: '/discovery/custom/01VAN_INST-vandev/html/homepage/report_broken_link.html'
});

app.controller('prmAlmaViewitAfterController', [ function() {
    var vm = this;
    
    vm.tcc_dialog_content = "<form id='tcc_bad_link_form' method='post'>";
    
     try {
		vm.tcc_info = vm.parentCtrl.item.pnx.addata;
		for(var property_name in vm.tcc_info) {
			if(property_name!='abstract') {
				vm.tcc_dialog_content += '<input type="hidden" id="' + property_name + '" name="' + property_name + '" value="' + vm.tcc_info[property_name] + '">';
			}
		}
		
		if(undefined!=vm.parentCtrl.item.pnx.addata.atitle) {
			vm.tcc_dialog_content += "<div class='tcc-dialog-title'><div style='color:green;font-weight:bold;'>" + vm.parentCtrl.item.pnx.addata.atitle + "</div></div>";
		} else if(undefined!=vm.parentCtrl.item.pnx.addata.btitle) {
			vm.tcc_dialog_content += "<div class='tcc-dialog-title'><div style='color:green;font-weight:bold;'>" + vm.parentCtrl.item.pnx.addata.btitle + "</div></div>";
		} else if(undefined!=vm.parentCtrl.item.pnx.addata.jtitle) {
			vm.tcc_dialog_content += "<div class='tcc-dialog-title'><div style='color:green;font-weight:bold;'>" + vm.parentCtrl.item.pnx.addata.jtitle + "</div></div>";
		}
		
		if(undefined!=vm.parentCtrl.item.pnx.addata.au) {
			vm.tcc_dialog_content += "<div class='tcc-dialog-au'>by " + vm.parentCtrl.item.pnx.addata.au + "</div>";
		} else {
			if(undefined!=vm.parentCtrl.item.pnx.addata.aulast) {
				vm.tcc_dialog_content += "<div class='tcc-dialog-aulast'>by " + vm.parentCtrl.item.pnx.addata.aulast+"</div>";
				
				if(undefined!=vm.parentCtrl.item.pnx.addata.aufirst) {
					vm.tcc_dialog_content += "<div class='tcc-dialog-aufirst'>" + vm.parentCtrl.item.pnx.addata.aufirst+"</div>";
				}
			}
		}
		
		if(undefined!=vm.parentCtrl.item.pnx.addata.jtitle) {
			vm.tcc_dialog_content += "<div class='tcc-dialog-jtitle' style='font-weight:bold;'>" + vm.parentCtrl.item.pnx.addata.jtitle + "</div>";
		}
		
		if(undefined!=vm.parentCtrl.item.pnx.addata.eissn) {
			vm.tcc_dialog_content += "<div class='tcc-dialog-eissn'>ISSN: " + vm.parentCtrl.item.pnx.addata.eissn + "</div>";
		} else if(undefined!=vm.parentCtrl.item.pnx.addata.issn) {
			vm.tcc_dialog_content += "<div class='tcc-dialog-issn'>ISSN: " + vm.parentCtrl.item.pnx.addata.issn + "</div>";
		}
		
		if(undefined!=vm.parentCtrl.item.pnx.addata.volume) {
			vm.tcc_dialog_content += "<div class='tcc-dialog-volume'>Vol: " + vm.parentCtrl.item.pnx.addata.volume + "</div>";
		}
		
		if(undefined!=vm.parentCtrl.item.pnx.addata.issue) {
			vm.tcc_dialog_content += "<div class='tcc-dialog-issue'>Issue: " + vm.parentCtrl.item.pnx.addata.issue + "</div>";
		}
		
		if(undefined!=vm.parentCtrl.item.pnx.addata.pages) {
			vm.tcc_dialog_content += "<div class='tcc-dialog-pages'>Pages: " + vm.parentCtrl.item.pnx.addata.pages + "</div>";
		} else if(undefined!=vm.parentCtrl.item.pnx.addata.spage) {
			vm.tcc_dialog_content += "<div class='tcc-dialog-pages'>Pages: " + vm.parentCtrl.item.pnx.addata.spage;
			if(undefined!=vm.parentCtrl.item.pnx.addata.epage) {
				vm.tcc_dialog_content += "- " + vm.parentCtrl.item.pnx.addata.epage;
			}
			vm.tcc_dialog_content += "</div>";
		}
		
		var w = window.innerWidth || document.documentElement.clientWidth || document.body.clientWidth;
		if(w<768) {
			vm.tcc_dialog_content += "<div style='border:1px solid #CECECE;padding:8px;'><div><strong>Which of following best describes the problem with this article?</strong></div><div class='tcc-dialog-problem'><select name='badlink_report_option_id' id='badlink_report_option_id' class='' size='1'><option value=''>-- select one --</option><option value='1' >PDF is blank/missing pages</option><option value='2' >404/page not found error</option><option value='3' >Article requires payment</option><option value='4' >Links to wrong page</option><option value='5' >Full text not available</option><option value='6' >Something else (explain below)</option></select></div></div>";
		} else {
			vm.tcc_dialog_content += "<div style='border:1px solid #CECECE;padding:8px;'><div><strong>Which of following best describes the problem with this article?</strong></div><div class='tcc-dialog-problem'><select name='badlink_report_option_id' id='badlink_report_option_id' class='' size='1'><option value=''>-- select one --</option><option value='1' >The PDF is blank/missing pages</option><option value='2' >I received a 404/page not found error</option><option value='3' >The website prompted me to pay to access the article</option><option value='4' >The link went to another website other than the selected article</option><option value='5' >Full text for the article was not available, only the abstract or citation</option><option value='6' >Something else went wrong, explain in the comments below</option></select></div></div>";
		}
		vm.tcc_dialog_content += "<div style='border:1px solid #CECECE;padding:8px;'><div><strong>We are often able to find the article using other means.<br>If we locate this item, to what <span style='font-style:italic;color:blue;'>email address</span> should we send the article? (Optional)<br>(Your email address will only be used to send you the article.)</strong></div><div class='tcc-dialog-email'><b>Email address:</b> <input type='text' id='badlink_report_email' name='badlink_report_email' value='' style='width:80%;border:1px solid #ECECEC;'></div></div>";
		vm.tcc_dialog_content += "<div style='border:1px solid #CECECE;padding:8px;'><div><label for='cc_email'><input type='checkbox' id='cc_email' name='cc_email' value='1'><strong>Send a copy of this report to the email above. (Optional)</strong></label></div></div>";
		vm.tcc_dialog_content += "<div style='border:1px solid #CECECE;padding:8px;'><div><strong>Comments (Optional)</strong></div><div class='tcc-dialog-comment'><textarea id='badlink_report_comments' name='badlink_report_comments' style='width:98%;'></textarea></div></div>";
		vm.tcc_dialog_content += "<input type='hidden' id='vid' name='vid' value='" + vm.parentCtrl.fullViewService.configurationUtil.searchFieldsService._searchParams.vid + "'>";
		vm.tcc_dialog_content += "<input type='hidden' id='tab' name='tab' value='" + vm.parentCtrl.fullViewService.configurationUtil.searchFieldsService._tab + "'>";
		vm.tcc_dialog_content += "<input type='hidden' id='docid' name='docid' value='" + vm.parentCtrl.item.pnx.control.recordid + "'>";
		vm.tcc_dialog_content += "<input type='hidden' id='context' name='context' value='" + vm.parentCtrl.item.context + "'>";
		vm.tcc_dialog_content += "<input type='hidden' id='search_scope' name='search_scope' value='" + vm.parentCtrl.fullViewService.configurationUtil.searchFieldsService._scope + "'>";
		vm.tcc_dialog_content += "<input type='hidden' id='badlink_report_location' name='badlink_report_location' value='" + window.location.href + "'>";
		vm.tcc_dialog_content += "<input type='hidden' id='lang' name='lang' value='en'></form>";
		
		document.getElementsByClassName("tingle-modal-box__content")[0].innerHTML = vm.tcc_dialog_content;
	} catch(err){
		console.log(err);
	}

	vm.getText = getText;
	
	function getText() {
		return "Report a broken link";
	}
	
}]);

	/** Report a Broken Link **/


/** End Report a Problem **/





/**  POST ANGULAR **/
/** Tingle for Report a Problem **/
/* Tingle JavaScript plugin */
function(t,o){"function"==typeof define&&define.amd?define(o):"object"==typeof exports?module.exports=o():t.tingle=o()}(this,function(){function t(t){var o={onClose:null,onOpen:null,beforeOpen:null,beforeClose:null,stickyFooter:!1,footer:!1,cssClass:[],closeLabel:"Close",closeMethods:["overlay","button","escape"]};this.opts=r({},o,t),this.init()}function o(){this.modalBoxFooter&&(this.modalBoxFooter.style.width=this.modalBox.clientWidth+"px",this.modalBoxFooter.style.left=this.modalBox.offsetLeft+"px")}function e(){this.modal=document.createElement("div"),this.modal.classList.add("tingle-modal"),0!==this.opts.closeMethods.length&&-1!==this.opts.closeMethods.indexOf("overlay")||this.modal.classList.add("tingle-modal--noOverlayClose"),this.modal.style.display="none",this.opts.cssClass.forEach(function(t){"string"==typeof t&&this.modal.classList.add(t)},this),-1!==this.opts.closeMethods.indexOf("button")&&(this.modalCloseBtn=document.createElement("button"),this.modalCloseBtn.classList.add("tingle-modal__close"),this.modalCloseBtnIcon=document.createElement("span"),this.modalCloseBtnIcon.classList.add("tingle-modal__closeIcon"),this.modalCloseBtnIcon.innerHTML="×",this.modalCloseBtnLabel=document.createElement("span"),this.modalCloseBtnLabel.classList.add("tingle-modal__closeLabel"),this.modalCloseBtnLabel.innerHTML=this.opts.closeLabel,this.modalCloseBtn.appendChild(this.modalCloseBtnIcon),this.modalCloseBtn.appendChild(this.modalCloseBtnLabel)),this.modalBox=document.createElement("div"),this.modalBox.classList.add("tingle-modal-box"),this.modalBoxContent=document.createElement("div"),this.modalBoxContent.classList.add("tingle-modal-box__content"),this.modalBox.appendChild(this.modalBoxContent),-1!==this.opts.closeMethods.indexOf("button")&&this.modal.appendChild(this.modalCloseBtn),this.modal.appendChild(this.modalBox)}function s(){this.modalBoxFooter=document.createElement("div"),this.modalBoxFooter.classList.add("tingle-modal-box__footer"),this.modalBox.appendChild(this.modalBoxFooter)}function i(){this._events={clickCloseBtn:this.close.bind(this),clickOverlay:l.bind(this),resize:this.checkOverflow.bind(this),keyboardNav:n.bind(this)},-1!==this.opts.closeMethods.indexOf("button")&&this.modalCloseBtn.addEventListener("click",this._events.clickCloseBtn),this.modal.addEventListener("mousedown",this._events.clickOverlay),window.addEventListener("resize",this._events.resize),document.addEventListener("keydown",this._events.keyboardNav)}function n(t){-1!==this.opts.closeMethods.indexOf("escape")&&27===t.which&&this.isOpen()&&this.close()}function l(t){-1!==this.opts.closeMethods.indexOf("overlay")&&!d(t.target,"tingle-modal")&&t.clientX=t},t.prototype.checkOverflow=function(){this.modal.classList.contains("tingle-modal--visible")&&(this.isOverflow()?this.modal.classList.add("tingle-modal--overflow"):this.modal.classList.remove("tingle-modal--overflow"),!this.isOverflow()&&this.opts.stickyFooter?this.setStickyFooter(!1):this.isOverflow()&&this.opts.stickyFooter&&(o.call(this),this.setStickyFooter(!0)))},{modal:t);
/* create a tingle modal dialog */ var tcc_modal = new tingle.modal({
   footer: true,
   stickyFooter: false,
   closeMethods: ['overlay', 'button', 'escape'],
   closeLabel: "Close",
   cssClass: ['custom-class-1', 'custom-class-2'],
   onOpen: function() {
      //console.log('modal open');
   },
   onClose: function() {
      //console.log('modal closed');
   },
   beforeClose: function() {
      return true; // close the modal
   }
});
tcc_modal.addFooterBtn('Send Report', 'tingle-btn tingle-btn--primary', function() {
   var form_data = new FormData(document.getElementById("tcc_bad_link_form"));
   var xhttp = new XMLHttpRequest();
   xhttp.onreadystatechange = function() {
      if (this.readyState == 4 && this.status == 200) {
         alert(this.responseText);
         return true;
      }
   };
   xhttp.open("POST", "https://your.domain.edu/path_to_code/", true);
   //xhttp.setRequestHeader("Content-Type", "application/x-www-form-urlencoded");
   xhttp.send(form_data);
   tcc_modal.close();
});
